glib/src/prelude.rs

```diff
@@ -4,6 +4,16 @@
 //! Traits and essential types intended for blanket imports.
 
 pub use crate::{
-    param_spec::ParamSpecBuilderExt, Cast, CastNone, Continue, IsA, ObjectExt, ObjectType,
-    ParamSpecType, StaticType, StaticTypeExt, StaticVariantType, ToSendValue, ToValue, ToVariant,
+    closure::{IntoClosureReturnValue, TryFromClosureReturnValue},
+    collections::{ptr_slice::IntoPtrSlice, strv::IntoStrV},
+    error::ErrorDomain,
+    gstring::{IntoGStr, IntoOptionalGStr},
+    object::{Cast, CastNone, IsA, ObjectExt, ObjectType},
+    param_spec::{HasParamSpec, ParamSpecBuilderExt, ParamSpecType},
+    types::{StaticType, StaticTypeExt},
+    value::{
+        ToSendValue, ToValue, ToValueOptional, ValueType, ValueTypeChecker, ValueTypeOptional,
```

Comment:
`ToValueOption` / `ValueTypeOptional` should be unnecessary here. Probably also others, I'd have to check each in detail.

Generally, the idea is that the prelude only contains the traits of which users would call methods directly (so they can call them without importing the trait explicitly). Exception for this are the FFI traits (so `IntoStrV` etc need some thinking).

Other traits that are only used in trait bounds (exception `IsA`) are not part of the prelude.

Reply:
> Generally, the idea is that the prelude only contains the traits of which users would call methods directly (so they can call them without importing the trait explicitly). Exception for this are the FFI traits (so IntoStrV etc need some thinking).

I am sure i have to make use of  `ToValue` somewhere. Probably before the enum/flags paramspec builder taking a default value param.

For IntoStrV and other ffi traits, should we re-export them in translate then? That would make things much cleaner then exporting everything in the main crate lib.rs and ending up with a very very long list of types/traits/functions

Reply:
> I am sure i have to make use of `ToValue` somewhere.

`ToValue` yes but not `ToValueOptional`. `ToValue` should stay in the prelude

> For IntoStrV and other ffi traits, should we re-export them in translate then

Yes that's what I was thinking

Reply:
Cleaned things a little bit, should be good for a review now